### ISSUE TITLE
json field related errors

### ISSUE THREAD

**Issue by SOHELAHMED7** [cebe/yii2-openapi#109]
Lets say I have json fields 

```
... other fields

message1
message2
message3
message4

... other fields

Gii/Api generation and Migrations of these fields are already applied

```

example:

```
message1:
          type: object
          x-db-type: JSON
          default: '{}'
```

Now if I add 3 new fields : message5, message6, message7 after message4 then alter column statement for message{1-4} fields is generated in migrations which should not be generated

```php
public function up()
{
    $this->addColumn('{{%table}}', 'message4', 'json NOT NULL AFTER message4');
    $this->addColumn('{{%table}}', 'message5', 'json NOT NULL');
    $this->addColumn('{{%table}}', 'message6', 'json NOT NULL');

    $this->alterColumn('{{%table}}', 'message2', 'json NOT NULL');
    $this->alterColumn('{{%table}}', 'message3', 'json NOT NULL');
    $this->alterColumn('{{%table}}', 'message1', 'json NOT NULL');
    $this->alterColumn('{{%table}}', 'message4', 'json NOT NULL');
}

public function down()
{
    $this->alterColumn('{{%table}}', 'message4', 'json NOT NULL');
    $this->alterColumn('{{%table}}', 'message1', 'json NOT NULL');
    $this->alterColumn('{{%table}}', 'message3', 'json NOT NULL');
    $this->alterColumn('{{%table}}', 'message2', 'json NOT NULL');

    $this->dropColumn('{{%table}}', 'message6');
    $this->dropColumn('{{%table}}', 'message5');
    $this->dropColumn('{{%table}}', 'message4');
}
```

TODO:

 - [x] check for default values in json e.g. `'{}'` and `[ ]`
 - [x] add tests for x-db-type (for json): fresh, secondary with new col; secondary with edit col

**Comment by SOHELAHMED7** [comment]
This issue can be closed. Fixed by #112 / https://github.com/SOHELAHMED7/yii2-openapi/pull/2

**Comment by SOHELAHMED7** [comment]
Reopening as the PR is not yet merged in origin repo
